katello/hostname-change.rb

```diff
@@ -11,17 +11,20 @@ module KatelloUtilities
   class HostnameChange
     include ::KatelloUtilities::Helper
 
-    def initialize(proxy, plural_proxy, proxy_hyphenated, program=nil, scenario=nil, accepted_scenarios=nil)
+    def initialize(init_options)
       @default_program = self.get_default_program
-      @proxy = proxy
-      @plural_proxy = plural_proxy
-      @proxy_hyphenated = proxy_hyphenated
-      @accepted_scenarios = accepted_scenarios
-
+      @proxy = init_options.fetch(:proxy)
+      @plural_proxy = init_options.fetch(:plural_proxy)
+      @proxy_hyphenated = init_options.fetch(:proxy_hyphenated)
+      @command_prefix = init_options.fetch(:command_prefix)
+      # accepted_scenarios is not a required parameter, so we allow to return nil.
+      # See method in helper file
+      @accepted_scenarios = init_options[:accepted_scenarios]
       @last_scenario = self.last_scenario
+
       @options = {}
-      @options[:program] = program || @default_program
-      @options[:scenario] = scenario || @last_scenario
+      @options[:program] = init_options[:program].fetch(:program, @default_program)
+      @options[:scenario] = init_options[:scenario].fetch(:scenario, @last_scenario)
```

Comment:
I think you mean to leave off the [:...] for these two options?

Reply:
updated